fswatcher/watch.go

```diff
@@ -0,0 +1,59 @@
+package fswatcher
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/fsnotify/fsnotify"
+)
+
+// Watch for changes to a directory on the filesystem and sends a notification to eventCh every time a file in the folder is changed.
+// Although it's possible to watch for individual files, that's not recommended; watch for the file's parent folder instead.
+// Note that changes are batched for 0.5 seconds before notifications are sent
+func Watch(ctx context.Context, dir string, eventCh chan<- struct{}) error {
+	watcher, err := fsnotify.NewWatcher()
+	if err != nil {
+		return fmt.Errorf("failed to create watcher: %w", err)
+	}
+	defer watcher.Close()
+
+	err = watcher.Add(dir)
+	if err != nil {
+		return fmt.Errorf("watcher error: %w", err)
+	}
+
+	batchCh := make(chan struct{}, 1)
+	defer close(batchCh)
+
+	for {
+		select {
+		// Watch for events
+		case event := <-watcher.Events:
+			if event.Op&fsnotify.Create == fsnotify.Create ||
+				event.Op&fsnotify.Write == fsnotify.Write {
+				if strings.Contains(event.Name, dir) {
+					// Batch the change
+					select {
+					case batchCh <- struct{}{}:
+						go func() {
+							time.Sleep(500 * time.Millisecond)
+							eventCh <- struct{}{}
```

Comment:
Should we also read from `batchCh` here to "reset" it? Since `batchCh` has a buffer size of 1, once it gets a `struct{}{}`, it will always go into `default`, even after the notification after 500ms is sent to `eventCh`.

Reply:
Yes, you're right. Thanks for catching that, will fix it

Reply:
Fixed